lsp/nls/src/contracts.rs

```diff
@@ -0,0 +1,290 @@
+//! Configuration for the Nickel Language Server
+use lsp_types::Url;
+use nickel_lang_core::term::IndexMap;
+use notify_debouncer_full::{
+    notify::{EventKind, RecommendedWatcher, RecursiveMode},
+    DebouncedEvent, Debouncer, RecommendedCache,
+};
+use serde::Deserialize;
+
+use std::{
+    collections::{HashMap, HashSet},
+    path::{Path, PathBuf},
+    sync::{Arc, RwLock},
+    time::Duration,
+};
+
+use crate::files::uri_to_path;
+
+pub const CONFIG_FILE_NAME: &str = "Nls-contracts.ncl";
+
+/// Contract configuration for nls.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct ContractConfig {
+    pub contract_path: PathBuf,
+}
+
+/// Contract configuration for nls.
+#[derive(Debug, Deserialize)]
+pub struct ContractConfigFormat {
+    /// This can be a relative or absolute path. If relative, it's relative to
+    /// the local config file that this belongs to.
+    contract_path: PathBuf,
+}
+
+impl ContractConfigFormat {
+    fn relative_to(self, parent_path: &Path) -> ContractConfig {
+        ContractConfig {
+            contract_path: parent_path.join(self.contract_path),
+        }
+    }
+}
+
+/// A collection of file-local nls configurations, provided by a single
+/// config file.
+///
+/// When we find a config file (called `nls.ncl`) in the filesystem,
```

Comment:
Should `nls.ncl` be updated here? Or is it a separate config file, in addition to the contract configuration?